particle_counter/main/hmi.m

```diff
@@ -352,22 +355,28 @@ function pushbutton1_Callback(hObject, eventdata, handles)
     if ~exist(['dataset_evaluation\', TodayDate],'dir')
         mkdir('dataset_evaluation\', TodayDate)
     end
+     strSystemTime = m2xdate(now);
+%      contamination_hist(:, Flag.Time) = [strSystemTime, calculation.contam_level_gram_per_liter, calculation.contam_level_num_per_sample];
+    histogram = []; 
+    for k = 1:20
+         histogram = [histogram sum(calculation.equiv_diam<k/10 & calculation.equiv_diam>(k-1)/10)];
+    end
+     contamination_hist(:, Flag.Time) = [strSystemTime, calculation.contam_level_gram_per_liter, calculation.contam_level_num_per_sample, histogram];
     
-%    strSystemTime = floor(clock);
-    strSystemTime = now;
-%    strSystemTime = [int2str(strSystemTime(4)) ':' int2str(strSystemTime(5)) ':' int2str(strSystemTime(6))];
-%     contamination_hist(:,Flag.Time) = [Flag.Time; calculation.contam_level_gram_per_liter];
-    contamination_hist(:,Flag.Time) = [strSystemTime; calculation.contam_level_gram_per_liter];
-    
-    cmap = colormap('gray');
-    if isunix
+     cmap = colormap('gray');
+     if isunix
 %         save(['dataset_evaluation/', TodayDate, '/', filename], 'calculation', 'raw_image', 'cropped_raw_image');
         calculation_noimage = rmfield(calculation,{'raw_image','cropped_raw_image'});
         save(['dataset_evaluation/', TodayDate, '/', filename], 'calculation_noimage');
         imwrite(calculation.raw_image,         cmap, ['dataset_evaluation/', TodayDate, '/', num2str(Flag.Time), '.jpg'], 'jpeg');
         imwrite(calculation.cropped_raw_image * 256, cmap, ['dataset_evaluation/', TodayDate, '/', num2str(Flag.Time), 'C.jpg'], 'jpeg');
-        fileID = fopen(['dataset_evaluation/', TodayDate, '/', 'contamination.txt'],'w');
-        fprintf(fileID,'%8f %12.8f\r\n', contamination_hist);
+ 
+        saveScatter = getframe(handles.axes7);
+        saveScatter = frame2im(saveScatter);
+        imwrite(saveScatter,['dataset_evaluation/', TodayDate, '/', num2str(Flag.Time), 'S.jpg'], 'jpeg');
```

Comment:
It was only saving the cropped images, the scatters representing particle were not saved.